src/harness/testcases/WINNF_FT_S_MES_testcase.py

```diff
@@ -0,0 +1,150 @@
+
+import json
+import os
+import unittest
+import sas
+from util import winnforum_testcase
+from datetime import datetime
+
+
+class MeasurementTestcase(unittest.TestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_MES_2(self):
+    """The sas under test to request measurement reporting
+    Register cbsds 6 CBSDs with measCapability make sure the SAS request 
+    The measurement report config after for heartbeat.
+    """
+
+    # Load the devices
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    device_f = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_f.json')))
+    device_g = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_g.json')))
+    device_i = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_i.json')))
+    devices = [device_a, device_c, device_e, device_f, device_g, device_i]
+
+    for device in devices:
+        device['measCapability'] = ['RECEIVED_POWER_WITH_GRANT']
+        # Inject FCC and user IDs
+        self._sas_admin.InjectFccId({'fccId': device['fccId']})
+        self._sas_admin.InjectUserId({'userId': device['userId']})
+
+    device_f['measCapability'] = ['RECEIVED_POWER_WITHOUT_GRANT']
+    cbsd_ids = []
+
+    # Register devices
+    request = {'registrationRequest': devices}
+    response = self._sas.Registration(request)['registrationResponse']
+    # Check registration response
+    self.assertEqual(len(response), 6)
+    for resp in response:
+        self.assertTrue('cbsdId' in resp)
+        self.assertEqual(resp['response']['responseCode'], 0)
+        self.assertFalse('RECEIVED_POWER_WITH_GRANT' in resp['measReportConfig'])
+        cbsd_ids.append(resp['cbsdId'])
+    del request, response
+    # Request grant
+    grant_request = []
+    for cbsd_id in cbsd_ids:
+        grant = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant['cbsdId'] = cbsd_id
+        grant_request.append(grant)
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+
+    grant_ids = []
+    # Check grant response
+    for response_num, resp in enumerate(response):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertTrue('grantId' in resp)
+        self.assertEqual(resp['response']['responseCode'], 0)
+        grant_ids.append(resp['grantId'])
+    del request, response
+    
+    # Trigger to request measurement report for all subsequent heartbeat request
+    self._sas_admin.TriggerMeasurementReportHeartbeat({'measReportConfig':[]})
```

Comment:
TriggerMeasurementReportHeartbeat does not have any inputs [documented in the testing API doc under Section: 'Trigger SAS to always request measurement reports in Heartbeat response'], so it ought to be an empty call.